vscode/vscode/vscode.ml

```diff
@@ -2107,10 +2171,57 @@ module Progress = struct
   include [%js: val report : t -> value:value -> unit [@@js.call]]
 end
 
+module TextDocumentContentChangeEvent = struct
+  include Interface.Make ()
+
+  include
+    [%js:
+    val range : t -> Range.t [@@js.get]
+
+    val rangeLength : t -> int [@@js.get]
+
+    val rangeOffset : t -> int [@@js.get]
+
+    val text : t -> string [@@js.get]]
+end
+
+module TextDocumentChangeEvent = struct
+  include Interface.Make ()
+
+  include
+    [%js:
+    val contentChanges : t -> TextDocumentContentChangeEvent.t list [@@js.get]
+
+    val document : t -> TextDocument.t [@@js.get]]
+end
+
+module TextDocumentContentProvider = struct
+  include Interface.Make ()
+
+  module OnDidChange = Event.Make (Uri)
+
+  include
+    [%js:
+    val onDidChange : t -> OnDidChange.t [@@js.get]
```

Comment:
Possible undefined. Should return `OnDidChange.t or_undefined`.

https://github.com/microsoft/vscode/blob/1.58.2/src/vs/vscode.d.ts#L1619

Reply:
Also done in db5b8077e4f4e909438c050c4b4b8813884fc6f9